### PR TITLE
Feature/interactive physical layout

### PR DESCRIPTION
**Pull request by jbrekelmans**
1/3 Pull requests on repositories Netflix/vizceral, Netflix/vizceral-example and Netflix/vizceral-react with the following features around "optimizing layout manually":
1. Repositioning of nodes by dragging them
2. Interactive Particle System simulation (nodes are particles and edges are spring forces) because it's awesome and allow for good automatic layout on some graphs
3. Resetting layout if #2 get's out of hand
4. Layout algorithm that respects positional metadata in the JSON graph files and otherwise lays out nodes in a circle
5. Export (well, Console.log) current node positions to be used as positional metadata once a satisfactory manual layout has been created

How to run:
1. Clone the branch feature/interactive_physical_layout of repositories jbrekelmans/vizceral, jbrekelmans/vizceral-example and jbrekelmans/vizceral-react locally
2. Run command 'npm link' in vizceral
3. Run command 'npm link vizceral' in vizceral-react
4. Run command 'npm link' in vizceral-react
5. Run command 'npm link vizceral-react' in vizceral-example
6. Make sure a graph (e.g. in sample_data.json) with 10-20 nodes is loaded for best results using the default Particle System settings

### CODE REVIEW

File: src/components/trafficFlow.jsx [hunk 1]
```diff
@@ -246,6 +324,16 @@ class TrafficFlow extends React.Component {
     this.setState({ displayOptions: displayOptions });
   }
 
+  physicsOptionsChanged = (physicsOptions) => {
+    this.setState({ currentGraph_physicsOptions: physicsOptions });
+    let currentGraph = this.state.currentGraph;
+    if (currentGraph == null) currentGraph = null;
+    Console.debug('TrafficFlow.jsx physicsOptionsChanged: ', physicsOptions, currentGraph, this.vizcReactComp);
```
Comment (jrsquared) [inline src/components/trafficFlow.jsx]: Are all these Console.debug messages just leftover from development? I don't see a need for these to be active in master...

**Comment by jolexa** [comment]
Very exciting feature.

**Comment by jrsquared** [comment]
There's a lot going on in here with your custom `backend` strategy and your own specific implementation of 'history'.  This example app was purposefully left as close to a simple example of using the component as possible with making the least number of implementation assumptions.

The physics options dropdown is great, adding the node movability option to the menu, and even adding a sample ring layout graph to the sample_data would be plenty for this vizceral-example app.

If you want to keep hacking on extra features and fake backends, I would be happy to link to your fork as a more advanced example.

**Comment by jbrekelmans** [comment]
@jrsquared I agree, the backend/history stuff should be in a separate branch and is not really part of what this pull request was originally intended for. I'll work on splitting it.

**Comment by jbrekelmans** [comment]
Finished/tested

**Pull request merged**